### DISCUSSION TITLE
Attack/Defend Issues

### DISCUSSION THREAD

**Discussion by tillmas** [tillmas/dcsgroundai#11] [Ideas]
Given the way that DCS ground unit AI works, there is no difference between what we are calling an attack zone and a defense zone.  We need to make a decision whether we are going to continue to differentiate between these two, as it has a lot of implications:

1.  We need to keep track of zone control.
2.  We need to have some way of outputting zone control at the end of the mission.
3.  We need to have some way of providing input to the AI commander on which units are better suited for attack and defense.

This all seems like unnecessary complexity from the perspective of a DCS pilot trying to enjoy a mission.  

I would appreciate any thoughts you have on this subject, my vote would be to keep it simple.

**Comment by tillmas** [comment]
Thinking about this, what if we had the mission designer consider three types of zones:
Blue Zones
Red Zones
Zones that were not to be considered for the ground AI mechanic

They could distinguish at start by name, and we could use that simply for:
1.  dropping defending infantry in some zones
2. making things happen on the map

I'm now thinking that this is a "sidecar".  It doesn't really affect the logic of the zone allocation or force movement, but it does give us a nice ability to do some cool stuff.   It could all run in parallel with everything else that we are doing.

**Comment by agridenour** [comment]
Agree on keeping it simple. I think we can just create the zone-typing function, but not implement any special logic for it at this point.
